### DISCUSSION TITLE
Is there a way to disable API access?

### DISCUSSION THREAD

**Discussion by siyavash4812** [supabase/supabase#4869] [Questions] [answered]
Hi,
I am new to this library and trying to understand how it works. 

I want to do some validation before doing updates/inserts/delete on the records in the table so I have decided to use my own backend server (expressJs) and then use the supabase client.

Is there a way to disable supabase API routes?

**Comment by steve-chavez** [comment, marked as answer]
> I want to do some validation before doing updates/inserts/delete on the records in the table 

You have many options for doing those on the database: SQL constraints, domains, types, triggers, RLS policies, functions. An additional backend server shouldn't be necessary.

> Is there a way to disable supabase API routes?

Not possible for now.

**Reply by siyavash4812** [reply]
Thank you for your answer. I am coming from mongodb background so I didnt know about triggers, constraints etc. Thank you 👍

**Comment by alexghattas** [comment]
@steve-chavez  what would you recommend being the best way to disable all API access to Supabase from the client? We are using Supabase with an existing backend, and would like to completely disable the API from the client.

I know we can set policies on each table to do so, but that only prevents the user from getting results, it still returns a 200.

Is there any way to globally disable the API, rather then setting individual policies for every table?

**Reply by steve-chavez** [reply]
One way would be to not have your database objects in the `public` schema, as that's what the API exposes.

Another way would be to remove the `public` schema from the API settings. which is not possible right now. I suggest you ask that as a feature request on the supabase repo.

Note that the API cannot be fully disabled since Storage depends on it.

**Reply by alexghattas** [reply]
Thanks Steve, thats a great solution to just rename the schema!

**Comment by Malki** [comment]
Was facing the same issue, I found the idea of revoking the anon key privileges discussed here helpful:
https://github.com/orgs/supabase/discussions/4547

**Comment by abdirahmn1** [comment]
Supabase now has the ability to "unexpose" the `public` schema, go to [settings -> api](https://supabase.com/dashboard/project/_/settings/api) and remove the `public` schema, after that go ahead and disable the `pg_graphql` extension. done, now your database cannot be queried using `supabase-js` :rocket: :+1:
